src/test/javascript/spec/component/statistics/statistics-graph.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { Graphs, SpanType } from 'app/entities/statistics.model';
+import { MockTranslateService } from '../../helpers/mocks/service/mock-translate.service';
+import { of } from 'rxjs/internal/observable/of';
+import * as moment from 'moment';
+import { SimpleChange } from '@angular/core';
+import { ChartsModule } from 'ng2-charts';
+import { HttpTestingController } from '@angular/common/http/testing';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('StatisticsGraphComponent', () => {
+    let fixture: ComponentFixture<StatisticsGraphComponent>;
+    let component: StatisticsGraphComponent;
+    let service: StatisticsService;
+    let httpMock: HttpTestingController;
+
+    beforeEach(fakeAsync(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, RouterTestingModule.withRoutes([]), MomentModule, ChartsModule],
+            declarations: [StatisticsGraphComponent, MockComponent(StatisticsGraphComponent), MockPipe(TranslatePipe)],
+            providers: [
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(StatisticsGraphComponent);
+                component = fixture.componentInstance;
+                service = TestBed.inject(StatisticsService);
+                httpMock = TestBed.get(HttpTestingController);
+            });
+    }));
+
+    afterEach(fakeAsync(() => {
+        jest.clearAllMocks();
+    }));
+
+    it('should initialize', fakeAsync(() => {
+        let graphData: number[] = [];
+        component.graphType = Graphs.SUBMISSIONS;
+        let arrayLength = 0;
+        const spy = spyOn(service, 'getChartData');
+
+        for (const span of Object.values(SpanType)) {
+            component.currentSpan = span;
+            switch (span) {
+                case SpanType.DAY:
+                    arrayLength = 24;
+                    break;
+                case SpanType.WEEK:
+                    arrayLength = 7;
+                    break;
+                case SpanType.MONTH:
+                    const startDate = moment().subtract(1, 'months');
+                    arrayLength = moment().diff(startDate, 'days');
+                    break;
+                case SpanType.QUARTER:
+                    arrayLength = 12;
+                    break;
+                case SpanType.YEAR:
+                    arrayLength = 12;
+                    break;
+            }
+            graphData = [];
+            for (let i = 0; i < arrayLength; i++) {
+                graphData[i] = i + 1;
+            }
+            spy.and.returnValue(of(graphData));
+
+            component.ngOnInit();
+            tick();
```

Comment:
This shouldn't need to be async, in the other tests we use `fixture.detectChanges()` which already takes care of `ngOnInit` too.

Reply:
I tried to use detectChanges() to trigger the ngOnInit as well, however, after the first iteration of the SpanTypes, the `dataForSpanType` attribute which is set in the ngOnInit call is not updated anymore. Calling ngOnInit() explicitly solved the problem.

Reply:
Weird. Is that still a problem without the `async` stuff?

Reply:
Yeah, still occurs without async calls